netlogo-gui/src/main/ide/ShowUsageBox.scala

```diff
@@ -0,0 +1,168 @@
+import java.awt.event._
+import javax.swing.table.{DefaultTableCellRenderer, DefaultTableModel, TableCellRenderer}
+import javax.swing._
+
+import org.nlogo.core.{Femto, Token, TokenType, TokenizerInterface}
+import org.nlogo.editor.{EditorArea, HighlightEditorKit}
+
+class ShowUsageBox(editorArea: EditorArea) {
+
+  val usageBox = new JDialog()
+  val dataModel = new DefaultTableModel(){
+    override def isCellEditable(row: Int, column: Int): Boolean = false
+    override def getColumnClass(columnIndex: Int): Class[_] = {
+      if(columnIndex == 0) classOf[Token] else classOf[String]
+    }
+  }
+  val usageTable = new JTable(dataModel)
+  val scrollPane = new JScrollPane(usageTable)
+  usageBox.add(scrollPane)
+  usageBox.setUndecorated(true)
+  usageTable.setDefaultRenderer(classOf[Token], new LineNumberRenderer())
+  usageTable.setDefaultRenderer(classOf[String], new LineRenderer(None))
+
+  usageBox.addWindowFocusListener(new WindowFocusListener {
+    override def windowLostFocus(e: WindowEvent): Unit = {
+      usageBox.setVisible(false)
+      dataModel.synchronized(dataModel.setRowCount(0))
+    }
+
+    override def windowGainedFocus(e: WindowEvent): Unit = {}
+  })
+  dataModel.setColumnCount(2)
+  usageTable.setSelectionMode(javax.swing.ListSelectionModel.SINGLE_SELECTION)
+  usageTable.getColumnModel.getColumn(0).setMaxWidth(40)
+  usageTable.getColumnModel.setColumnSelectionAllowed(false)
+  usageTable.getTableHeader().setReorderingAllowed(false)
+  usageTable.setTableHeader(null)
+  usageTable.setRowMargin(3)
+  usageTable.setRowHeight(20)
+  usageTable.getColumnModel.getColumn(1).setMinWidth(200)
+  usageTable.getColumnModel.getColumn(0).setMinWidth(40)
+  usageTable.setFont(editorArea.getFont)
+
+  usageTable.addMouseListener(new MouseAdapter() {
+    override def mouseClicked(e: MouseEvent): Unit = {
+      usageBox.setVisible(false)
+      val token = usageTable.getValueAt(usageTable.getSelectedRow, 0).asInstanceOf[Token]
+      dataModel.synchronized(dataModel.setRowCount(0))
+      editorArea.select(token.start, token.end)
+    }
+  })
+
+  usageTable.addKeyListener(new KeyAdapter(){
+    override def keyPressed(e: KeyEvent): Unit = {
+      e.getKeyCode match {
+        case KeyEvent.VK_ESCAPE => usageBox.setVisible(false)
+        case _ => e.setSource(editorArea)
+          editorArea.dispatchEvent(e)
+      }
+    }
+  })
+
+  usageTable.setCursor(Cursor.getPredefinedCursor(Cursor.HAND_CURSOR))
+
+  def showBox(me: MouseEvent, position: Int): Unit = {
+    val tokenOption = findTokenContainingPosition(editorArea.getText(), position)
+    for {token <- tokenOption} {
+      if(token.tpe == TokenType.Ident || token.tpe == TokenType.Command || token.tpe == TokenType.Reporter) {
+        val tokens = getUsage(editorArea.getText(), token)
+        dataModel.setRowCount(0)
+        for (t <- tokens) {
+          dataModel.addRow(Array[AnyRef](t, editorArea.getLineText(t.start).trim))
```

Comment:
@mrerrormessage It looks OK to me(2 space indented). What do you exactly mean by weird?